common/dynamicconfig/constants.go

```diff
@@ -501,6 +501,16 @@ const (
 	AcquireShardInterval = "history.acquireShardInterval"
 	// AcquireShardConcurrency is number of goroutines that can be used to acquire shards in the shard controller.
 	AcquireShardConcurrency = "history.acquireShardConcurrency"
+	// ShardLingerEnabled configures if the shard controller will temporarily
+	// delay closing shards after a membership update, awaiting a shard
+	// ownership lost error from persistence. Not recommended with persistence
```

Comment:
The note here about "Not recommended" is due to shardLingerAndClose using AssertShardOwnership to try to generate a shard ownership lost error.